### ISSUE TITLE
🐛[BUG]: Rotating a Line gives the new line an interior.

### ISSUE THREAD

**Issue by ahallback** [NVIDIA/physicsnemo-sym#196]
### Version

1.6.0

### Describe the issue

Rotation of a Line() class somehow produces an interior for the line. It seems that the SDF might be calculated wrong for the rotated line. The interior seems to be sampled from the area between the x-axis and the new line. 

### Minimum reproducible example

```python
import numpy as np
from physicsnemo.sym.geometry.primitives_2d import Line
from physicsnemo.sym.utils.io.vtk import var_to_polyvtk

line = Line((0,0), (0, 42), normal=1)
line = line.rotate(angle=np.pi/2 + 0.1)
s = line.sample_interior(
        nr_points=1000,
    )
var_to_polyvtk(s, f"interior")
print("Volume: {:.3f}".format(np.sum(s["area"])))
```

**Comment by peterdsharpe** [comment]
Hey @ahallback, thanks for this report!

Basically what's happening here is that the interior points are sampled from an axis-aligned bounding box for the `Line` object. If the `Line` is itself axis-aligned (e.g., exactly horizontal or vertical), then the volume of the bounding box is correctly detected as zero, and hence no interior points are sampled. However, if the `Line` is rotated, then the axis-aligned bounding box has nonzero volume, and points are incorrectly sampled.

Ultimately, this is due to a limitation in the `Line()` class definition, as this class is only strictly valid for 2D lines parallel to the y-axis. From the constructor of `Line`:

```python

class Line(Geometry):
    """
    2D Line parallel to y-axis

    Parameters
    ----------
    point_1 : tuple with 2 ints or floats
        lower bound point of line segment
    point_2 : tuple with 2 ints or floats
        upper bound point of line segment
    normal : int or float
        normal direction of line (+1 or -1)
    parameterization : Parameterization
        Parameterization of geometry.
    """

    def __init__(self, point_1, point_2, normal=1, parameterization=Parameterization()):
        assert point_1[0] == point_2[0], "Points must have same x-coordinate"
        ... # and so on
```

In other words, this bug stems from the fact that a) `Line` was not written to allow rotation, but b) the fact that it extends the `Geometry` class allows a `.rotate()` method to be used.

Ultimately the immediate fix is to instead raise a `NotImplementedError` upon attempts to rotate `Line`; longer-term, pending user demand, we may be able to extend the `Line` class to correctly work with rotations by using a different bounding box parameterization.

Action item: PR to overwrite `Line.rotate()` with `NotImplementedError`, and checks to ensure this doesn't break tests.
